proposed/2021/PackageNamespaces.md

```diff
@@ -102,33 +106,433 @@ If a user would like to support namespaces on different package sources, they ca
+
+```xml
+<!-- Project 2 -->
+<PackageReference Include="Microsoft.B" Version="1.0.0" />
+```
+
+**Result:**
+
+- Not deterministic, because it depends on the order in which the projects are restored, this could lead to either package getting restored from either source.
+This is *not* a very common, nor a recommended scenario.
+
 ## Drawbacks
 
 <!-- Why should we not do this? -->
-There are many ways to exclude & include dependencies in various CI/CD providers like Azure DevOps, GitHub Actions, and more. These are typically for individual dependencies or upstream packages. This however is not supported universally for each CI/CD provider & can be confusing on the limitations. 
+There are many ways to exclude & include dependencies in various CI/CD providers like Azure DevOps, GitHub Actions, and more. These are typically for individual dependencies or upstream packages. This however is not supported universally for each CI/CD provider & can be confusing on the limitations.
+
+Package namespaces has best security practices in mind at the cost of user experience. One thing NuGet is known for is "just working". By using package namespaces, we are functionally changing how you might think about including new dependencies into your software supply chain.
 
-The migration for users to leverage package namespaces will be tedious as they will need to lookup each package ID / namespace used in their project(s). There are no tools planned to do this on behalf of the user & therefore will be mostly a one-time manual cost.
+The migration for users to leverage package namespaces will be tedious as they will need to lookup each package ID / namespace used in their project(s). There are no tools planned to do this on behalf of the user & therefore will be a one-time manual effort.
 
 ## Rationale and alternatives
 
 <!-- Why is this the best design compared to other designs? -->
 <!-- What other designs have been considered and why weren't they chosen? -->
 <!-- What is the impact of not doing this? -->
 
-We believe that this feature combined with [Source Pinning] will make users more secure using NuGet than they have ever been. Given that other ecosystems like npm & Maven support a concept of scoped or banned dependencies through a concept of namespaces, NuGet would largely benefit from this type of feature.
+We believe that this feature provides the highest degree of control and allow users to be more secure using NuGet than they have ever been. Given that other ecosystems like npm & Maven support a concept of scoped or banned dependencies through a concept of namespaces, NuGet would largely benefit from this type of feature as it complements NuGet's existing ability of [reserving package namespaces](https://docs.microsoft.com/nuget/nuget-org/id-prefix-reservation). This is a benefit for any internal packages in which a [best practice is having a package prefix](https://docs.microsoft.com/nuget/create-packages/package-authoring-best-practices#package-id).
```

Comment:
How does this feature provide any benefit if customers already use a single feed as source in their nuget.config with multiple upstreams in that single feed? Looks like this feature has the intended benefit only when multiple feeds are used in nuget.config.

Reply:
The benefits mostly exist for multiple package sources(I.e public/private/local sources). If you're already on a single feed, the additional benefit you have is limiting the namespaces from that single feed into your projects/solutions.